### ISSUE TITLE
Filter state properties like filtering components

### ISSUE THREAD

**Issue by ndabAP** [vuejs/vue-devtools#401]
**Problem**: I have a complex and deep nested Vuex store. I often track those nested properties. The problem is now, every edit to the store requires a full page reload and the opened Vuex state is gone.

**Feature request**: It would be cool to search the store for properties like you can search components to quickly watch the desired nested property.

Might be a duplicate of https://github.com/vuejs/vue-devtools/issues/259, but the referenced issue with the referenced merge request didn't fulfill the feature request.

**Comment by LinusBorg** [comment]
I just have two sidenotes:

1. Hot Relaoding

> The problem is now, every edit to the store requires a full page reload and the opened Vuex state is gone.

Have you actually applied the code outlined in vuex's docs for hot reloading? full page refreshs should not be necessary, usually.

https://vuex.vuejs.org/en/hot-reload.html

2. Deeply nesting store

This sis generally not advised. Data should be normalized if you can. I realize that this is not always possible and your situation might be one of those, but I thought to mention it.

**Comment by ndabAP** [comment]
> Have you actually applied the code outlined in vuex's docs for hot reloading?

Nice, something I'll definitely try out.

> I realize that this is not always possible

Exactly. Imagine you have shop with a MongoDB database with redundancy and, like I do, one entity is mapped to one module. You may have something like this:

```
user: {
  basket: {
    products: {[
      product: {
        fromUser: {
          name: ''
        }
      }
    }]
  }
}
```

**Comment by ndabAP** [comment]
@yyx990803 this issue is not resolved. Or do you consider it implicitly resolved through #418 

The behavior I except would the same as filtering components: The match jumps to the deepest possibly child node. Like:

```
User
  > Products
    > Product
```

Becomes when searching for `Product`:

```
Product
```

**Comment by Akryum** [comment]
Except it doesn't work like that. 😛 

```
Products
  > Product
```
